File: app/src/main/java/com/example/cleanarchitecture_toyproject/viewmodel/AndroidApplication.java
package com.example.cleanarchitecture_toyproject.viewmodel;

import android.app.Application;

public class AndroidApplication extends Application {

    private static AndroidApplication instance;
    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }
    public static AndroidApplication getInstance(){
        return instance;
    }
}

File: app/src/main/java/com/example/cleanarchitecture_toyproject/viewmodel/AndroidApplication.kt
package com.example.cleanarchitecture_toyproject.viewmodel

import android.app.Application
import android.content.Context

class AndroidApplication : Application() {
   /* override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {

        var instance: AndroidApplication? = null
            private set

        operator fun get(context: Context) : AndroidApplication {
            return context.applicationContext as AndroidApplication
        }
    }*/

    init {
        INSTANCE = this
    }

    override fun onCreate() {
        super.onCreate()
    }

    companion object {
        lateinit var INSTANCE : AndroidApplication
    }
}
